CSS: Negative descriptor (#29717)
mdn/content: files/en-us/web/css/@counter-style/negative/index.md

{{CSSRef}}

The **`negative`** descriptor of the {{cssxref("@counter-style")}} at-rule lets you define how negative counter values are represented when defining custom counter styles. The value of the `negative` descriptor defines the symbols to be prepended and appended to the counter representation when the counter's value is negative.

> ## Syntax

Reviewer: We can improve the Syntax section:
```css
/* One <symbol> value */
negative: "-"; /* Adds '-' before if counter value is negative */

/* Two <symbol> values */
negative: "(" ")"; /* Adds '(' before and ')' after if counter value is negative */
```

### Rendering negative counters

> This example [extends](/en-US/docs/Web/CSS/@counter-style/system#extends) the [`decimal`](/en-US/docs/Web/CSS/list-style-type#decimal) list style, adding a `-` and parenthesis around negative counter values.

Reviewer: ```suggestion
This example [extends](/en-US/docs/Web/CSS/@counter-style/system#extends) the [`decimal`](/en-US/docs/Web/CSS/list-style-type#decimal) list style. The `negative` descriptor is used to add `(-` and `)` before and after negative counter values.
```

#### HTML

```html
<ol start="-3">

>   <li>negative three</li>
>   <li>negative two</li>
>   <li>negative one</li>
>   <li>zero</li>
>   <li>one</li>

Reviewer: Wouldn't it be better to keep the original example? If we go with the updated one, we can make minor updates:

```suggestion
  <li>Negative three</li>
  <li>Negative two</li>
  <li>Negative one</li>
  <li>Zero</li>
  <li>One</li>
```

</ol>
```

#### CSS

```css
@counter-style neg {
  system: extends decimal;

>   negative: "(-" ")";

Reviewer: Should we add a `suffix` here:
```suggestion
  negative: "(-" ")";
  suffix: ": ";
```

}

ol {
  list-style: neg;
}

> ```

Reviewer: We can add a line after the "Result" frame to say that `0` and `1` are not surrounded by the `negative` descriptor symbols because the counter values are not negative.

Author: Added:

> The prefix and suffix listed as the value of the `negative` descriptor are only added to the marker when the counter value is less than zero.

## See also

> - Other {{cssxref("@counter-style")}} descriptors, including, {{cssxref("@counter-style/system","system")}}, {{cssxref("@counter-style/symbols", "symbols")}}, {{cssxref("@counter-style/additive-symbols", "additive-symbols")}}, {{cssxref("@counter-style/prefix", "prefix")}}, {{cssxref("@counter-style/suffix", "suffix")}}, {{cssxref("@counter-style/range", "range")}}, {{cssxref("@counter-style/pad", "pad")}}, {{cssxref("@counter-style/speak-as", "speak-as")}}, and {{cssxref("@counter-style/fallback", "fallback")}}
> - {{Cssxref("list-style")}}, {{Cssxref("list-style-image")}}, {{Cssxref("list-style-position")}}
> - {{cssxref("symbols", "symbols()")}}, the functional notation creating anonymous counter styles.

Reviewer: ```suggestion
- {{cssxref("@counter-style")}} descriptors: {{cssxref("@counter-style/system","system")}}, {{cssxref("@counter-style/symbols", "symbols")}}, {{cssxref("@counter-style/additive-symbols", "additive-symbols")}}, {{cssxref("@counter-style/prefix", "prefix")}}, {{cssxref("@counter-style/suffix", "suffix")}}, {{cssxref("@counter-style/range", "range")}}, {{cssxref("@counter-style/pad", "pad")}}, {{cssxref("@counter-style/speak-as", "speak-as")}}, {{cssxref("@counter-style/fallback", "fallback")}}
- List style properties: {{Cssxref("list-style")}}, {{Cssxref("list-style-image")}}, {{Cssxref("list-style-position")}}
- {{cssxref("symbols", "symbols()")}} function to create anonymous counter styles
```
